test/terra/states/test_aer_statevector.py

```diff
@@ -0,0 +1,1154 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019, 2020, 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Integration Tests for AerStatevector
+"""
+
+import unittest
+from math import pi
+import numpy as np
+import logging
+from itertools import permutations
+from ddt import ddt, data
+from numpy.testing import assert_allclose
+
+from qiskit import transpile
+from qiskit.exceptions import QiskitError
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+from qiskit.providers.basicaer import QasmSimulatorPy
+from qiskit.quantum_info.random import random_unitary, random_statevector, random_pauli
+from qiskit.quantum_info.states import Statevector
+from qiskit.circuit.library import QuantumVolume
+from qiskit.providers.aer import AerSimulator
+from qiskit.quantum_info.operators.operator import Operator
+from qiskit.quantum_info.operators.symplectic import Pauli, SparsePauliOp
+from qiskit.quantum_info.operators.predicates import matrix_equal
+from qiskit.visualization.state_visualization import numbers_to_latex_terms, state_to_latex
+from qiskit.circuit.library import QFT, HGate
+from test.terra import common
+from qiskit_aer.aererror import AerError
+from qiskit_aer.quantum_info.states import AerStatevector
+
+
+logger = logging.getLogger(__name__)
+
+@ddt
+class TestAerStatevector(common.QiskitAerTestCase):
+    """AerState tests"""
```

Comment:
It would be good to provide some more stress tests of the memory allocation, moving and de-allocation routines, since there's quite a lot of complexity in there.